### DISCUSSION TITLE
Should we Provide spark-daria sql functions API in snake_case instead of camelCase ?

### DISCUSSION THREAD

**Discussion by zeotuan** [MrPowers/spark-daria#170] [Ideas]
Hi everyone,

I wanted to discuss whether we should switch the Spark-Daria API to use snake_case instead of camelCase for daria functions API.

**Context:**
- Scala generally follows camelCase conventions.
- Spark, despite being written in Scala, uses snake_case for its SQL functions API

**Pros for camelCase:**
- Idiomatic : Aligns with Scala conventions and would feel more natural for Scala developers.
- Consistency with Scala libraries: Other Scala libraries primarily use camelCase, making it consistent with the broader ecosystem.

**Pros for snake_case:**
- Consistency with Spark SQL functions API, provide seamless and intuitive user experience
- Reduced Confusion and cognitive load for user

**suggestion:**
- start providing new daria sql function api as snake_case
- providing snake_case alias for existing function

**Note**
This does not affect other API such as DataframeExt API which is generally only use camelCase even in pyspark

**Comment by MrPowers** [comment]
I am cool with whatever you and @alfonsorr think is best, but let's make sure we have the best practices documented here once we make a decision: https://github.com/mrpowers-io/spark-style-guide/blob/main/SPARK_SCALA_STYLE_GUIDE.md

**Reply by alfonsorr** [reply]
If we really want to allow it i would say both. As you already said the scala spark API it's a Frankenstein monster. To have an alias it's not a problem.
